src/main/resources/db/migration/0400/V403__InternalUsersModified.sql

```diff
@@ -0,0 +1,39 @@
+ALTER TABLE project_internal_users
+    ADD COLUMN created_time TIMESTAMP WITH TIME ZONE DEFAULT NOW();
+ALTER TABLE project_internal_users
+    ADD COLUMN modified_time TIMESTAMP WITH TIME ZONE DEFAULT NOW();
+
+ALTER TABLE project_internal_users
+    ADD COLUMN created_by BIGINT REFERENCES users;
+ALTER TABLE project_internal_users
+    ADD COLUMN modified_by BIGINT REFERENCES users;
+
+-- Most internal users have been Project Leads, which includes TF Contacts
+UPDATE project_internal_users piu
+SET (created_by, modified_by, created_time, modified_time) = (
+    SELECT ou.created_by, ou.modified_by, ou.created_time, ou.modified_time
+    FROM organization_users ou
+    WHERE ou.user_id = piu.user_id
+    AND ou.organization_id = (select organization_id from projects where id = piu.project_id)
+    AND ou.role_id = 5 -- TF CONTACT
+)
+WHERE piu.project_internal_role_id in (1, 2) -- only do this for users that became TF Contacts through internal roles
+;
+
+-- If any others remain, use system user
+UPDATE project_internal_users piu
+SET (created_by, modified_by, created_time, modified_time) = (
```

Comment:
Since the time columns default to `NOW()`, you could skip setting them here.

Reply:
Ya it doesn't like that, and I can't set not null and default in the same statement (plus the default already exists). I think the default is only used when inserting after not null has been enforced.
> ERROR: column "created_time" of relation "project_internal_users" contains null values

Reply:
Interesting, this discovered cases where users have `Project Lead` but not TF Contact. Looking into this